test/e2e/page-objects/pages/confirmations/redesign/gas-fee-modal.ts

```diff
@@ -0,0 +1,305 @@
+   *
+   * @param option - The gas option to check
+   * @param expectedValue - The expected value text
+   */
+  async checkGasOptionValue(
+    option: 'low' | 'medium' | 'high' | 'advanced' | 'site_suggested',
+    expectedValue: string,
+  ): Promise<void> {
+    console.log(
+      `Checking if ${option} gas option shows value: ${expectedValue}`,
+    );
+    const selector = `[data-testid="gas-option-${option}"]`;
+    await this.driver.waitForSelector({
+      css: selector,
+      text: expectedValue,
+    });
+  }
+
+  // ============ Advanced EIP-1559 Modal Methods ============
+
+  /**
+   * Checks if the advanced EIP-1559 modal is displayed.
+   */
+  async checkAdvancedEIP1559ModalIsDisplayed(): Promise<void> {
+    console.log('Checking if advanced EIP-1559 modal is displayed');
+    await this.driver.waitForSelector(this.advancedEIP1559Modal);
+  }
+
+  /**
+   * Enters the max base fee value in the advanced EIP-1559 modal.
+   *
+   * @param value - The max base fee value to enter (in Gwei)
+   */
+  async enterMaxBaseFee(value: string): Promise<void> {
+    console.log(`Entering max base fee: ${value}`);
+    await this.driver.fill(this.maxBaseFeeInput, value);
+  }
+
+  /**
+   * Enters the priority fee value in the advanced EIP-1559 modal.
+   *
+   * @param value - The priority fee value to enter (in Gwei)
+   */
+  async enterPriorityFee(value: string): Promise<void> {
+    console.log(`Entering priority fee: ${value}`);
+    await this.driver.fill(this.priorityFeeInput, value);
+  }
+
+  /**
+   * Enters the gas limit value.
+   *
+   * @param value - The gas limit value to enter
+   */
+  async enterGasLimit(value: string): Promise<void> {
+    console.log(`Entering gas limit: ${value}`);
+    await this.driver.fill(this.gasLimitInput, value);
+  }
+
+  // ============ Advanced Gas Price Modal Methods (Legacy) ============
+
+  /**
+   * Checks if the advanced gas price modal (legacy) is displayed.
+   */
+  async checkAdvancedGasPriceModalIsDisplayed(): Promise<void> {
+    console.log('Checking if advanced gas price modal is displayed');
+    await this.driver.waitForSelector(this.advancedGasPriceModal);
+  }
+
+  /**
+   * Enters the gas price value in the legacy advanced modal.
+   *
+   * @param value - The gas price value to enter (in Gwei)
+   */
+  async enterGasPrice(value: string): Promise<void> {
+    console.log(`Entering gas price: ${value}`);
+    await this.driver.fill(this.gasPriceInput, value);
+  }
+
+  // ============ Common Modal Actions ============
+
+  /**
+   * Clicks the save button to save gas fee changes.
+   */
+  async clickSave(): Promise<void> {
+    console.log('Clicking save button');
+    await this.driver.clickElement(this.saveButton);
+  }
+
+  /**
+   * Clicks the cancel button to go back to estimates modal.
+   */
+  async clickCancel(): Promise<void> {
+    console.log('Clicking cancel button');
+    await this.driver.clickElement(this.cancelButton);
```

Comment:
could we do this to prevent race conditions and make the subsequent action more stable


```suggestion
    await this.driver.clickElementAndWaitToDisappear(this.cancelButton);
```

Reply:
Fixed in https://github.com/MetaMask/metamask-extension/pull/38857/changes/01e712f81942f8fe3aa43a6a4e26274e82abceca